src/objects/core/models.py

```diff
@@ -62,25 +62,9 @@ class Object(models.Model):
 
     objects = ObjectQuerySet.as_manager()
 
-    def get_actual_record(self, date):
-        return (
-            self.records.filter(start_at__lte=date)
-            .filter(models.Q(end_at__gte=date) | models.Q(end_at__isnull=True))
-            .order_by("-pk")
-            .first()
-            # TODO: pk should prolly be index once added.
-        )
-
-    def get_registration_record(self, date):
-        return (
-            self.records.filter(registration_at__lte=date)
-            .order_by("-registration_at")
-            .first()
-        )
-
     @property
     def current_record(self):
-        return self.get_actual_record(datetime.date.today())
+        return self.records.filter_for_date(datetime.date.today()).first()
 
     @property
     def last_record(self):
```

Comment:
Shouldn't the last record be indicated simply with:

```
self.records.order_by("-index").first()
```

?

Reply:
> I thought that you can create a record that would start in the future, in this case, it wouldn't be retrieved here.

True, but is that the goal of this function? You only use `last_record` in the `ObjectRecord.save` method (as far as I can tell):

```python
    def save(self, *args, **kwargs):
        if not self.id and self.object.last_record:
            self.index = self.object.last_record.index + 1

            #  add end_at to previous record
            previous_record = self.object.last_record
            previous_record.end_at = self.start_at
            previous_record.save()

        super().save(*args, **kwargs)
```
 As for:

```python
        if not self.id and self.object.last_record:
            self.index = self.object.last_record.index + 1
```

The idea is that every record has an index value in the order they were added to the system, right? If you take `start_at` into account this can mess up things.

```python
            previous_record = self.object.last_record
            previous_record.end_at = self.start_at
            previous_record.save()
```

Here, it's different and I wonder if this is correct. If I correct a previous record, should the previous record then be adjusted?

Reply:
Yeah, you are right, I'll merge `current_record` and `last_record` together into one property that displays the record with the maximum index.

About corrections: we don't take them into account when displaying objects and their records, and we don't validate `start_date` and `end_date` of correcting records. Perhaps we need to think over how corrections should work and perhaps ask stakeholders?
Questions from the top of my head:
* could correcting record have `start_date` different from the correcred record
* if yes (and in StuF it's possible) should we change `end_date` of the record before the corrected one
* If they have different `start_date` it is possible that we will show the corrected record in the list endpoint. Is it ok or we need to exclude corrected records from being displayed?

Reply:
As discussed, we'll leave this as is and do the rest in #168